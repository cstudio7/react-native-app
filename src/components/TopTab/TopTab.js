import React from 'react';
import { Text, TouchableOpacity, View, StyleSheet } from 'react-native';

export default class TopTab extends React.Component {
  render() {
    const { tabs } = this.props;
    return (
      <View style={styles.tabsContainer}>
        {tabs.map((tab,index) => {
          return (
            <TouchableOpacity
            key={tab.name + index}
              onPress={tab.onPress}
              style={[styles.tab, tab.isActive ? styles.tabActive : null]}>
              <Text style={styles.tabText}>{tab.name}</Text>
            </TouchableOpacity>
          );
        })}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  tabsContainer: {
    flexDirection: 'row',
    marginBottom: 15
  },
  tab: {
    flex: 1,
    height: 40,
    alignItems: 'center',
    paddingVertical: 11,
    borderBottomWidth: 2,
    borderBottomColor: '#fff'
  },
  tabActive: {
    borderBottomColor: '#000'
  },
  tabText: {
    fontWeight: '600',
    textAlign: 'center'
  }
});
