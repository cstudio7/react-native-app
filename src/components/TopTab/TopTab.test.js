import React from 'react';
import renderer from 'react-test-renderer';

import TopTab from './TopTab';

const mockedTabs = Object.freeze([
  {
    name: 'Женские имена',
    onPress: () => {},
    isActive: true
  },
  {
    name: 'Мужские имена',
    onPress: () => {},
    isActive: false
  }
]);

it('renders correctly', () => {
  const tree = renderer
    .create(<TopTab tabs={mockedTabs} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
