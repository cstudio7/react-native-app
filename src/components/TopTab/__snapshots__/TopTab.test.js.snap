// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 15,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#000",
        "borderBottomWidth": 2,
        "flex": 1,
        "height": 40,
        "opacity": 1,
        "paddingVertical": 11,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Женские имена
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomWidth": 2,
        "flex": 1,
        "height": 40,
        "opacity": 1,
        "paddingVertical": 11,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "600",
          "textAlign": "center",
        }
      }
    >
      Мужские имена
    </Text>
  </View>
</View>
`;
